omnibus/config/software/chefdk.rb

```diff
@@ -39,16 +39,12 @@
 
 relative_path "chef-dk"
 
-if windows?
-  dependency "ruby-windows"
-  dependency "ruby-windows-devkit"
-else
-  dependency "libffi" if debian?
-  dependency "ruby"
-end
-
-dependency "rubygems"
+dependency "ruby"
+dependency "rubygems-native"
 dependency "bundler"
+# Put this early in the dependency chain to ensure that nokogiri is compiled
+# correctly using the system libraries.
+dependency "nokogiri"
```

Comment:
It currently is but maybe the API doesn't guarantee it.  In any case, I don't think the order here actually matters as long as all the other software definitions we have correctly specify a dependency on nokogiri or libgecode if it's in the transitive closure of their gem dependencies.  I think most of them get it right?  It's a thing to double check later.